OpenRA.Mods.Common/Orders/RepairOrderGenerator.cs

```diff
@@ -12,86 +12,191 @@
 using System.Collections.Generic;
 using System.Linq;
 using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
 using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
 using OpenRA.Traits;
 
 namespace OpenRA.Mods.Common.Orders
 {
 	public class RepairOrderGenerator : OrderGenerator
```

Comment:
Would be better to let RepairOrderGenerator implement `IOrderGenerator` than inherit from ` OrderGenerator`. Then we dont need to make the change in `OrderGenerator.cs`

Reply:
I think probably make a new class `ChromeLogicOrderGenerator` for the `RepairOrderGenerator`, or even `PowerDownOrderGenerator and SellOrderGenerator` to herit can be better?

Reply:
Becuase `PowerDownOrderGenerator` can also apply dragging if later somebody want. We can let them change as less as possible of original code.